src/evict/evict_lru.c

```diff
@@ -1801,12 +1804,17 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
      */
     start = queue->evict_queue + *slotp;
     remaining_slots = max_entries - *slotp;
-    if (btree->evict_walk_progress >= btree->evict_walk_target) {
-        btree->evict_walk_target = __evict_walk_target(session);
+    if ((target_pages = __evict_walk_target(session)) == 0 ||
```

Comment:
Can you explain why you make this change?

Reply:
Sure @quchenhao 

In cases where trees are not suitable candidates, they should be skipped. 

A tree is marked unsuitable if it does not align with the expected eviction strategy. Specifically, if the strategy involves dirty eviction, trees containing only clean data should be skipped, as their pages would not be selected for eviction, leading to unnecessary work. The same applies to clean and update evictions. 

We already have the same implementation ([here](https://github.com/wiredtiger/wiredtiger/blame/7d0407c2904a4d1638478e459faa1157da4f8ef2/src/evict/evict_lru.c#L1746)) today. However, there are two issues with the current implementation: 
1. The suitability check is performed in __evict_walk_target, which is nested within a conditional that may not always be true. ([here](https://github.com/wiredtiger/wiredtiger/blame/7d0407c2904a4d1638478e459faa1157da4f8ef2/src/evict/evict_lru.c#L1804))
2. The check lacks a dedicated condition for update eviction. 

Addressing these two issues is the objective of this pull request.

Reply:
I think I understand the change. Thank you.

Reply:
It took me some time to understand why we add the check (target_pages = __evict_walk_target(session)) == 0 here. I think we are trying to disable the walk if we don't want any page. Can you add a comment for this?